build_cmake/helpers.cmake

```diff
@@ -420,17 +420,33 @@ function(config_lib config_name description)
     # Check that the configs dependencies are enabled before setting it to a visible enabled state.
     eval_dependency("${CONFIG_LIB_DEPENDS}" enabled)
     if(enabled)
+        message("-- Looking for library ${CONFIG_LIB_LIB}")
         # 'check_library_exists' won't use our current cache when test compiling the library.
         # To get around this we need to ensure we manually forward WT_ARCH and WT_OS as a minimum. This is particularly
         # needed if 'check_library_exists' will leverage one of our toolchain files.
         if((NOT "${WT_ARCH}" STREQUAL "") AND (NOT "${WT_ARCH}" STREQUAL ""))
             set(CMAKE_REQUIRED_FLAGS "-DWT_ARCH=${WT_ARCH} -DWT_OS=${WT_OS}")
         endif()
-        check_library_exists(${CONFIG_LIB_LIB} ${CONFIG_LIB_FUNC} "" has_lib_${config_name})
+        find_library(has_lib_${config_name} ${CONFIG_LIB_LIB})
+        #check_library_exists(${CONFIG_LIB_LIB} ${CONFIG_LIB_FUNC} "" has_lib_${config_name})
```

Comment:
We can remove this dead code :)

Reply:
I left that there on purpose for reference (along with the function names) in case it's wanted again. There are cases where a function might be in any of several base system libraries, or in libc and available by default, and in those cases you might want a version of config_lib() that does actually use the function name. (I was thinking that in the long run maybe there should be two versions of config_lib(), one that takes a function and one that doesn't, or something like that.) However, currently WT doesn't have any such cases where the function name matters, and maybe if any appear they should be handled by config_func() anyhow. I am not an expert on cmake, so if you think all of this should just go I'm happy to defer to your opinion; just wanted to give my reasoning first.

Reply:
Thanks for the background and reasoning on the change! That's completely fair & thanks for the consideration.

On second thought, to save the hassle, I'm cool to leave this code in just for know. I'll be adding some small cleanups in another upcoming PR that will probably involve using the `config_lib`  (with regards to the tcmalloc lib). If its still unnecessary then I will look into cleaning up the function :)

Reply:
If at that point you want me to do it, feel free to ping me...